### PR TITLE
feat: Middleware example to docs

### PR DESCRIPTION
**Pull request by casualmatt**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

New section with an example on how to use a middleware for auth redirect
https://github.com/Intevel/nuxt-directus/issues/8

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

**Comment by Intevel** [comment]
Thanks for this PR 💚
But please call it "Middleware example"

**Comment by casualmatt** [comment]
Fix 👍

File: docs/content/en/usage/useDirectusAuth.md
Comment (Intevel) [inline docs/content/en/usage/useDirectusAuth.md:103]: Change this and provide a full code example: 

```
<script setup lang="ts">
definePageMeta({
  middleware: ["auth"]
})
</script>
```

File: docs/content/en/usage/useDirectusAuth.md
Comment (Intevel) [inline docs/content/en/usage/useDirectusAuth.md:89]: Not composable, call it middleware

**Review by Intevel — CHANGES_REQUESTED** [review]
After this changes, I will approve. Thanks for your work 💚

**Comment by Intevel** [comment]
And you have to take a look at this: https://www.conventionalcommits.org/en/v1.0.0/
Fix your PR and commit messages, I cant merge it until you did.

**Comment by Intevel** [comment]
Fixed it, merged! Thanks for your work

**Comment by casualmatt** [comment]
Sorry, 
lunch time, next time i will follow your suggestion, so that you don't need to spent too much time on it. 
👍

**Comment by Intevel** [comment]
No problem

**Pull request merged**
